sound/soc/sof/intel/hda.c

```diff
@@ -77,17 +77,29 @@ int hda_ctrl_dai_widget_setup(struct snd_soc_dapm_widget *w, bool setup)
 		}
 
 		/* send DAI_CONFIG IPC */
-		return sof_ipc_tx_message(sdev->ipc, config->hdr.cmd, config, config->hdr.size,
+		ret = sof_ipc_tx_message(sdev->ipc, config->hdr.cmd, config, config->hdr.size,
 					  &reply, sizeof(reply));
+		if (ret < 0)
+			return ret;
+
+		sof_dai->configured = true;
 	}
 
+	/*
+	 * nothing to do if hw_free() is called without restarting the stream after resume
+	 */
+	if (!sof_dai->configured)
```

Comment:
Hmm, isn't this wrong now for setup=1 case. You set configured=1 on L85 and then you L91 check is false and we continue until sof_widget_free().

Reply:
oops missing the return after setting configured to true